pandas/io/packers.py

```diff
@@ -357,6 +357,7 @@ def encode(obj):
                     'klass': obj.__class__.__name__,
                     'axes': data.axes,
                     'blocks': [{'items': data.items.take(b.mgr_locs),
+                                'locs': b.mgr_locs.as_array,
```

Comment:
no, like you have it now is ok. for back-compat we can only add attributes (and can't change the meaning at all). So this is how I wanted it. The new code will decode using ``.locs`` as that avoids the duplicates problem, and can read the older msgpacks. This actually also is forward-compat in that a current version can be read by an older version (so this is the best case scenario). We don't have a tests for forward-compat, but you can manually verify (as would need the older code to run in the tests which is 'odd').